Pokedex Challenge/Pokedex Challenge/Views/View Controllers/PokemonDetailViewController.swift

```diff
@@ -0,0 +1,92 @@
+//
+//  PokemonDetailViewController.swift
+//  Pokedex Challenge
+//
+//  Created by Michael Flowers on 1/25/19.
+//  Copyright © 2019 Michael Flowers. All rights reserved.
+//
+
+import UIKit
+
+class PokemonDetailViewController: UIViewController, UISearchBarDelegate {
+
+    //MARK: - Properties
+    var pokemon: Pokemon? {
+        didSet {
+            updateViews()
+        }
+    }
+    var pokemonController: PokemonController?
+    
+    //MARK: - IBOutlets
+    @IBOutlet weak var searchBar: UISearchBar!
+    @IBOutlet weak var pokemonName: UILabel!
+    @IBOutlet weak var id: UILabel!
+    @IBOutlet weak var types: UILabel!
+    @IBOutlet weak var abilities: UILabel!
+    @IBOutlet weak var saveButtonProperties: UIButton!
+    
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        //set the searchBar delegate
+       searchBar.delegate = self
+    }
+    
+    //when user taps cell, updateViews when the view will appear again.
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        updateViews()
+    }
+    
+    //MARK: - PokemonController functions
+    @IBAction func savePokemon(_ sender: UIButton) {
+        navigationController?.popViewController(animated: true)
+    }
+    
+    //search bar delegate function
+    func searchBarSearchButtonClicked(_ searchBar: UISearchBar) {
+        //fetch the pokemon
+        
+        guard let name = searchBar.text, !name.isEmpty else { return }
+        pokemonController?.fetchPokemon(with: name, completion: { (error) in
+            if let error = error {
+                NSLog("Error fetching the pokemon from searchBar: \(error.localizedDescription)")
+                return
+            }
+            //this is where we append it to the array
+            DispatchQueue.main.async {
+                self.updateViews()
+                //I have to display the ui here.
+            }
+        })
+    }
+
+    //MARK: - Custom Functions
+    func updateViews(){
+        guard let passedInPoke = pokemon else {
+            
+            //no pokemon was passed in
+            pokemonName.isHidden = true
+            id.isHidden = true
+            types.isHidden = true
+            abilities.isHidden = true
+            saveButtonProperties.isHidden = true
+            title = "Pokemon Search"
+            return
+        }
+        
+        //pokemon was passed in
+        searchBar.isHidden = true
+        saveButtonProperties.isHidden = true
```

Comment:
You need to present all views to the user. So make their `isHidden` value false, if there is a pokemon. Use `map` to get the array and then `joined(separator: )` method to create a single string from the array
```suggestion
        saveButtonProperties.isHidden = false
        saveButtonProperties.setTitle("Save Pokemon", for: .normal)
        
        title = passedInPoke.name.capitalized
        
        id.isHidden = false
        id.text = "Id: " + String(passedInPoke.id)
        
        types.isHidden = false
        let typesArray = passedInPoke.types.map({ $0.type.name })
        types.text = "Type: " + typesArray.joined(separator: ", ")
        
        abilities.isHidden = false
        let abilityArray = passedInPoke.abilities.map({ $0.ability.name })
        abilities.text = "Ability: " + abilityArray.joined(separator: ", ")
        
        pokemonName.isHidden = false
        pokemonName.text = passedInPoke.name.capitalized
```